### ISSUE TITLE
Schedule Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. 

### ISSUE THREAD

**Issue by danielordonez** [primefaces/primeng#3088]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Current behavior**
Getting the following error with Schedule after upgrading to Angular 4:

`Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. `

**Expected behavior**
Same behavior as in Angular 2 and PrimeNG 2

**What is the motivation / use case for changing the behavior?**
Can't get schedule to work properly as it used to in angular 2

**Please tell us about your environment:**
MacOS Sierra VS Code

* **Angular version:** 4.2.2

* **PrimeNG version:** 4.0.3

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

**Comment by aurimukas** [comment]
+1

**Comment by fpalma-auge** [comment]
I also getting same error upgrading from 4.1.3 to 4.2.2

**Comment by Mrtcndkn** [comment]
Hi,

I tried it with PrimeNG 4.1.0-RC2 and Angular 4.2.2 and it looks fine.

    "@angular/animations": "4.2.2",
    "@angular/common": "4.2.2",
    "@angular/compiler": "4.2.2",
    "@angular/core": "4.2.2",
    "@angular/forms": "4.2.2",
    "@angular/http": "4.2.2",
    "@angular/platform-browser": "4.2.2",
    "@angular/platform-browser-dynamic": "4.2.2",
    "@angular/router": "4.2.2",
    "@angular/cli": "1.1.0",
    "@angular/compiler-cli": "4.2.2",
    "@angular/language-service": "4.2.2"

https://pasteboard.co/1m40j6e5L.png
Are there any other steps to reproduce the problem? @danielordonez

**Comment by danielordonez** [comment]
The issue happened until I cleaned and reinstalled the node modules of my local project, did you try cleaning everything and reinstalling the node modules? I just tried updating everything again and I'm getting the same issue even with PrimeNG 4.1.0 RC2 @Mrtcndkn

**Comment by Mrtcndkn** [comment]
Yes actually primeng is running with angular 4.1.3 so i cleaned it up and installed 4.2.3 it looks fine. Could you please create a plunkr about it?
http://plnkr.co/edit/Wj39h1?p=preview

**Comment by eXXeption** [comment]
Same error for me after upgrading to cli 1.0.4 and angular 4.2.2

**Comment by saverett** [comment]
Likely related to this issue in Angular: angular/angular#17572

**Comment by cagataycivici** [comment]
Yes, an issue with Angular.

please provide a test case based on plunkr below and we'll review again. Maybe we can do something.

http://plnkr.co/edit/Wj39h1?p=preview

**Comment by cagataycivici** [comment]
As we are unable to replicate, closing for now, once we have the plunkr from you, we'll review again.

**Comment by danielordonez** [comment]
I figured out that this happens when you store and display values from a variable that was assigned values from a Schedule method. For example, getting the date using the getDate() method, storing it in the variable "date" and then displaying "date" anywhere in the project.

I replicated the issue I had in this plunkr:

http://plnkr.co/edit/VPPWqS?p=preview

Just FYI, this is not happening in Angular 2 and PrimeNG 2

**Comment by danielordonez** [comment]
I've been following the issue and I noticed it was removed from the milestones it was previously in.

Any workaround for this issue? It's the only issue preventing us from upgrading our project to Angular 4

I appreciate it

**Comment by danielordonez** [comment]
It seems to be fixed by the latest version of PrimeNG 4.1.3 + Angular 4.3.4

Closing now

**Comment by Balaraju1** [comment]
I also getting same error after upgrading from 4.0.0 to 4.2.4
